system/jlib/jsocket.cpp

```diff
@@ -1164,6 +1164,7 @@ void CSocket::cancel_accept()
 #endif
```

Comment:
Might be good to add a comment that close() is not enough to trigger the poll() in wait_read() so we use a shutdown() here to get the poll() to return before its timeout period.